app.rb

```diff
@@ -1,34 +1,171 @@
+    @persons = []
+    @rentals = []
+  end
+
+  def create_teacher
+    puts '-creating a new teacher-'
+    print 'Name: '
+    name = gets.chomp
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'specialization: '
+    spec = gets.chomp
+    @persons.push(Teacher.new(age, spec, name))
+    puts "Person created successfully\n"
+  end
+
+  def permission?
+    print "Has parents' permission? [Y/N]: "
+    option = gets.chomp.capitalize
+    case option
+    when 'N'
+      false
+    when 'Y'
+      true
+    else
+      permission?
+    end
+  end
+
+  def create_student
+    puts '-creating a new student-'
+    print 'Name: '
+    name = gets.chomp
+    print 'Age: '
+    age = gets.chomp.to_i
+    permission = true
+    permission = permission? if age < 18
+    @persons.push(Student.new(age, name, parent_permission: permission))
+    puts "Person created successfully\n"
+  end
+
+  def create_person
+    print "\Do you want to create a student (1) or a teacher (2)? [Input the number]: "
+    option = gets.chomp.to_i
+    case option
+    when 1
+      create_student
+    when 2
+      create_teacher
+    else
+      create_person
+    end
+  end
+
+  def create_book
+    puts '-creating new book-'
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    @books.push(Book.new(title, author))
+  end
+
+  def select_book
+    puts 'Select a book from the list by [number]:'
+    @books.each_with_index { |book, i| puts "[#{i}] #{book.title} by #{book.author}" }
+    print 'Enter [number]: '
+    index = gets.chomp.to_i
+    return @books[index] if @books[index]
+
+    select_book
+  end
+
+  def select_person
+    puts 'Select a person from the list by [number]:'
+    @persons.each_with_index { |person, i| puts "[#{i}] [#{person.class.name}] ID: #{person.id} Name: #{person.name}" }
+    print 'Enter [number]: '
+    index = gets.chomp.to_i
+    return @persons[index] if @persons[index]
+
+    select_person
+  end
+
+  def create_rental
+    book = select_book
+    person = select_person
+    print 'insert date: '
+    date = gets.chomp
+    @rentals.push(Rental.new(date, book, person))
+    puts "rental created successfully.\n"
+  end
+
+  def all_books
+    @books.each_with_index { |book, i| puts "#{i + 1} - #{book.title} by #{book.author}" }
+  end
+
+  def all_people
+    @persons.each_with_index do |person, i|
+      puts "#{i + 1} - [#{person.class.name}] ID: #{person.id} Name: #{person.name}"
+    end
+  end
+
+  def person_rentals
+    puts "select person's [ID]:"
+    @persons.each { |person| puts "[#{person.id}] Name: #{person.name} [#{person.class.name}]" }
+    print 'Enter ID: '
+    id = gets.chomp.to_i
+    persona = @persons.find { |person| person.id == id }
+    if persona
+      persona.rentals.each do |rental|
+        puts "#{rental.date} - #{rental.person.name} - #{rental.book.title} by #{rental.book.author}"
+      end
+    else
+      puts "The id #{id} doesn't exist"
+    end
```

Comment:
- [ ] On listing the rentals there is an extra line that shouldn't be there, please 🙏🏽 correct this behavior to match with the original video.

![image](https://user-images.githubusercontent.com/58642949/226491866-82807075-e1bd-4220-a844-f378cb6a19ba.png)